aula-2/src/main/java/org/acelera/brasil/futebol/Jogador.java

```diff
@@ -0,0 +1,62 @@
+package org.acelera.brasil.futebol;
+
+/**
+ * Classe responsável pela representação de um jogador de futebol.
+ * 
+ * @author edmilson.santana
+ *
+ */
+public class Jogador {
+
+	private final String nome;
+
+	private final String cidade;
+
+	private final String pais;
+
+	private final Posicao posicao;
+
+	private final Integer gols;
+
+	Jogador(String nome, String cidade, String pais, Posicao posicao, Integer gols) {
+		this.nome = nome;
+		this.cidade = cidade;
+		this.pais = pais;
+		this.posicao = posicao;
+		if(gols < 0) {
```

Comment:
Pq não colocar esse regra dentro do Builder?
Uma vez que existe um builder não faz mais sentido ele garantir que o objeto seja criado válido?

Reply:
No caso das validações no Builder, isso não quebraria a integridade do próprio objeto, visto que ele pode ser criado diretamente sem uso do Builder?

Reply:
Se o contrutor não for público não será um problema para as classes fora do pacote.
Um outro ponto seria a criação de um inner class do builder e deixar o construtor do Jogador como privado